ADA/databricks_rstudio_sql_warehouse.qmd

```diff
@@ -1,105 +1,210 @@
+
+---
+
+## Pre-requisites
+
+Before you start, you must have:
+
+-   Access to the Databricks platform
 
--   Access to Databricks
--   Access to a SQL Warehouse on DataBricks
--   R and RStudio downloaded
+-   Access to data in a SQL Warehouse on Databricks
+
+-   R and RStudio downloaded and installed
+
+-   The `ODBC` and `DBI` packages installed in RStudio
+
+If you do not have access to Databricks or a SQL Warehouse within Databricks, you can request this using [a service request form](https://dfe.service-now.com/serviceportal?id=sc_cat_item&sys_id=74bc3be81b212d504f999978b04bcb0b).
+
+If you do not have R or RStudio, you can find them both in the Software Centre. Note that you need both R **and** RStudio installed.
+
+---
+
+## Process
+
+There are three steps to complete before your connection can be established. These are:
+
+-   Installing an ODBC driver on your laptop to enable a connection between your laptop and Databricks
+-   Modifying your .Renviron file to establish a connection between RStudio and Databricks
+-   Adding connection code to your existing scripts in RStudio
+
+Each of these steps is described in more detail in the sections below. 
+
+---
+
+### Setting up the ODBC driver
+
+An ODBC driver is required for the `ODBC` package in R to work - you must install it before attempting to use the package to connect to your data. 
+
+---
+
+#### Install the Simba Spark ODBC driver from the Software Centre
+
+---
+
+-   Open the Software Centre via the start menu
+
+-   In the 'Applications' tab, click `Simba Spark ODBC Driver 64-bit`
+
+::: {align="center"}
+![](../images/databricks-software-centre.png)
+:::
+
+-   Click install
 
 ------------------------------------------------------------------------
 
-# Downloading an ODBC driver
+### Establishing an RStudio connection using environment variables
 
 ------------------------------------------------------------------------
 
-1.  Install the 'Simba Spark ODBC' driver from the software centre.
+The `ODBC` package in RStudio allows you to connect to Databricks by creating and modifying four environment variables in your .Renviron file.
 
-    i)  Open the Software Centre via the start menu.
+To set the environment variables, call `usethis::edit_r_environ()`. You will then need to enter the following information:
 
-    ii) In the 'Applications' tab, click 'Simba Spark ODBC Driver 64-bit'. ![](../images/databricks-software-centre.png)
+```         
+DATABRICKS_HOST=<databricks-host>
+DATABRICKS_SQL_PATH=<sql-warehouse-path>
+DATABRICKS_TOKEN=<personal-access-token> 
+DATABRICKS_CLUSTER_ID=<personal-cluster-id>
+```
 
-    iii) Click install.
+::: callout-important
+Each value should be entered inside quotation marks, like this:
```

Comment:
What's the reason for this as it works either way?

Reply:
Ah I hadn't realised! If it works either way I'll remove this